src/Parser/Hunter/Marksmanship/Modules/Talents/CallingTheShots.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import SpellIcon from 'common/SpellIcon';
+import StatisticBox from 'Main/StatisticBox';
+import { formatNumber } from 'common/format';
+
+const COOLDOWN_REDUCTION_MS = 3000;
+
+/**
+ * Casting Arcane Shot or Multi-Shot reduces the cooldown of Trueshot by 3.0 sec.
+ */
+class CallingTheShots extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  effectiveTrueshotReductionMs = 0;
+  wastedTrueshotReductionMs = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.CALLING_THE_SHOTS_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.ARCANE_SHOT.id && spellId !== SPELLS.MULTISHOT_MM.id) {
+      return;
+    }
+    if (this.spellUsable.isOnCooldown(SPELLS.TRUESHOT.id)) {
+      if (this.spellUsable.cooldownRemaining(SPELLS.TRUESHOT.id) < COOLDOWN_REDUCTION_MS) {
+        const effectiveReductionMs = this.spellUsable.cooldownRemaining(SPELLS.TRUESHOT.id);
+        this.effectiveTrueshotReductionMs += effectiveReductionMs;
```

Comment:
You never actually reduce the cooldown here. 

Also you the above check doesn’t appear to be mecessary since without it the added wastedTrueshotReductionMs would be 0 anyway.